identity/src/routes/connect_group/associate_users.rs

```diff
@@ -48,7 +48,14 @@ impl crate::routes::Routes {
                 user_id, 
                 connect_group_id, 
                 user_role
-            ) "#,
+			) VALUES (
+				$1,
+				$2,
+				$3
+			)
+				ON CONFLICT (user_id, connect_group_id) DO UPDATE
+					SET user_role = EXCLUDED.user_role
+            "#,
```

Comment:
This wouldn't work because the query in this part we're using is built through a [QueryBuilder](https://docs.rs/sqlx/0.7.2/sqlx/struct.QueryBuilder.html)

Meaning that the values, etc. are pushed using the `push_values` and `push_bind` method. Is is why the query in the first place was only 

```rs
r#"INSERT INTO user_connect_group (
                user_id, 
                connect_group_id, 
                user_role
            ) "#
```

Take a look at this [example](https://docs.rs/sqlx/0.7.2/sqlx/struct.QueryBuilder.html#example-mysql) in the `sqlx` docs

Reply:
One thing that you could do is keep the query string the same as before. Then, use the `push` method to add the `ON CONFLICT` clause before calling the `build` method. 

which would look something like this:

```rs
sqlx::QueryBuilder::new(
            r#"INSERT INTO user_connect_group (
                user_id, 
                connect_group_id, 
                user_role
           ) "#,
        )
        .push_values(&body.users, |mut b, user| {
            b.push_bind(&user.user_id)
                .push_bind(&*id)
                .push_bind(&user.role_id);
        })
        .push(" ON CONFLICT (user_id, connect_group_id) DO UPDATE SET user_role = EXCLUDED.user_role")
        .build()
```